ui/static/ui/js/learning_resources.jsx

```diff
@@ -120,8 +127,65 @@ define('learning_resources', [
         _.filter(e.target.options, function(option) {
           return option.selected && option.value !== null;
         }), 'value');
-
-      this.props.setValues(this.props.selectedVocabulary.slug, selectedValues);
+      // check if the current vocabulary allows free tagging and in case add
+      // the new tags before proceeding
+      if (this.props.selectedVocabulary.vocabulary_type === 'f') {
+        // extract the list of terms not in the vocabulary existing terms
+        var termsSlugs = _.pluck(this.props.selectedVocabulary.terms, 'slug');
+        var termsToCreate = _.filter(selectedValues, function (slug) {
+          return _.indexOf(termsSlugs, slug) === -1;
+        });
+        // if there are no new terms to create, set the state
+        if (!termsToCreate.length) {
+          this.props.setValues(
+            this.props.selectedVocabulary.slug, selectedValues
+          );
+        } else {
```

Comment:
It is unlikely but what happens if a term is added and another term removed in the same event. Here the else condition doesn't handle the remove case.

This seems tricky since you would ideally want to call `setValues` once, otherwise one of the calls may overwrite the other call. You can do this with a promise, for example:
      var promises = _.map(termsToCreate, function(term) {
        return $.post() // post to create new term
      }
      return $.when.apply($, promises).then(function () {
        var args;
        if (promises.length === 1) {
          args = [arguments[0]];
        } else {
          args = arguments;
        }

        var newSlugs = ... // extract slugs from promise results in args
        thiz.props.setValues(thiz.props.selectedVocabulary.slug, newSlugs)
      });